DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/PerturbProcessor.cs

```diff
@@ -0,0 +1,136 @@
+            { DicomVR.SV, new VRTypes() { ElementType = typeof(DicomSignedVeryLong), ValueType = typeof(long[]) } },
+        };
+
+        public PerturbProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var perturbSetting = settingFactory.CreateAnonymizerSetting<PerturbSetting>(settingObject);
+            _perturbFunction = new PerturbFunction(perturbSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (item.ValueRepresentation == DicomVR.AS)
+            {
+                var values = ((DicomAgeString)item).Get<string[]>().Select(Utility.ParseAge).Select(x => _perturbFunction.Perturb(x));
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Select(Utility.AgeToString).ToArray());
+            }
+            else
+            {
+                if (!_numericValueTypeMapping.ContainsKey(item.ValueRepresentation))
+                {
+                    throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationFunction, $"Perturb is not supported for {item.ValueRepresentation}");
+                }
+
+                var elementType = _numericValueTypeMapping[item.ValueRepresentation].ElementType;
+                var valueType = _numericValueTypeMapping[item.ValueRepresentation].ValueType;
+
+                // Get numeric value using reflection.
+                var valueObj = elementType.GetMethod("Get").MakeGenericMethod(valueType).Invoke(item, new object[] { -1 });
+                PerturbNumericValue(dicomDataset, item, valueObj as Array);
+            }
+        }
+
+        private void PerturbNumericValue(DicomDataset dicomDataset, DicomItem item, Array values)
+        {
+            if (values.Length == 0)
+            {
+                return;
+            }
+
+            var valueType = values.GetValue(0).GetType();
+            if (valueType == typeof(decimal))
+            {
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Cast<decimal>().Select(_perturbFunction.Perturb).ToArray());
+            }
+            else if (valueType == typeof(double))
+            {
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Cast<double>().Select(_perturbFunction.Perturb).ToArray());
+            }
+            else if (valueType == typeof(float))
+            {
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Cast<float>().Select(_perturbFunction.Perturb).ToArray());
```

Comment:
@qfyin , is there an elegant way to express this?

Reply:
Seems needs refine, but have no idea how to make it elegant for now.